bioconda_utils/update_pinnings.py

```diff
@@ -1,88 +1,125 @@
-#!/usr/bin/env python
-from conda_build import api
 import re
 import sys
 import os.path
+import logging
+import collections
+import enum
+
+import networkx as nx
+
+from .utils import RepoData, load_conda_build_config, parallel_iter
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
 buildNumberRegex = re.compile("^( +number: +)(\d+)(.*?)$")
 buildNumberJinja = re.compile("^({% +set build +=.+?)(\d+)(.*?%})")
 
-def already_bumped(r, name, version, buildNumber):
-    rsub = r.loc[(r.name == name) &
-             (r.platform != "osx") &
-             (r.version == version)]
-    if rsub.shape[0] == 0:
-        # The version was never built, it doesn't explicitly require bumping
-        return True
-    if buildNumber > max(rsub.build_number):
-        # Already bumped
-        return True
-    return False
 
+def will_build_variant(meta):
+    """Check if the recipe variant will be built as currently rendered"""
+    build_numbers = RepoData().get_package_data(
+        'build_number',
+        name=meta.name(), version=meta.version(),
+        platform=['linux', 'noarch'],
+    )
```

Comment:
This says "get me `'build_number"` for packages where `name=meta.name()` AND ...".